internal/gcs-sidecar/handlers.go

```diff
@@ -128,13 +288,100 @@ func (b *Bridge) executeProcess(req *request) (err error) {
 	if err := commonutils.UnmarshalJSONWithHresult(req.message, &r); err != nil {
 		return fmt.Errorf("failed to unmarshal executeProcess: %w", err)
 	}
-
+	containerID := r.ContainerID
 	var processParams hcsschema.ProcessParameters
 	if err := commonutils.UnmarshalJSONWithHresult(processParamSettings, &processParams); err != nil {
 		return fmt.Errorf("executeProcess: invalid params type for request: %w", err)
 	}
 
-	b.forwardRequestToGcs(req)
+	commandLine := []string{processParams.CommandLine}
+
+	if containerID == UVMContainerID {
+		log.G(req.ctx).Tracef("Enforcing policy on external exec process")
+		_, _, err := b.hostState.securityPolicyEnforcer.EnforceExecExternalProcessPolicy(
+			req.ctx,
+			commandLine,
+			processParamEnvToOCIEnv(processParams.Environment),
+			processParams.WorkingDirectory,
+		)
+		if err != nil {
+			return errors.Wrapf(err, "exec is denied due to policy")
+		}
+		b.forwardRequestToGcs(req)
+	} else {
+		// fetch the container command line
+		c, err := b.hostState.GetCreatedContainer(req.ctx, containerID)
+		if err != nil {
+			log.G(req.ctx).Tracef("Container not found during exec: %v", containerID)
+			return errors.Wrapf(err, "containerID doesn't exist")
+		}
+
+		// if this is an exec of Container command line, then it's already enforced
+		// during container creation, hence skip it here
+		containerCommandLine := escapeArgs(c.spec.Process.Args)
+		if processParams.CommandLine != containerCommandLine {
+
+			user := securitypolicy.IDName{
+				Name: processParams.User,
+			}
+
+			log.G(req.ctx).Tracef("Enforcing policy on exec in container")
+			_, _, _, err = b.hostState.securityPolicyEnforcer.
+				EnforceExecInContainerPolicyV2(
+					req.ctx,
+					containerID,
+					commandLine,
+					processParamEnvToOCIEnv(processParams.Environment),
+					processParams.WorkingDirectory,
+					user,
+					nil,
+				)
+			if err != nil {
+				return errors.Wrapf(err, "exec in container denied due to policy")
+			}
+		}
+		headerID := req.header.ID
+
+		// initiate process ID
+		b.pendingMu.Lock()
+		b.pending[headerID] = nil // nil means not yet received
+		b.pendingMu.Unlock()
+
+		defer func() {
+			b.pendingMu.Lock()
+			delete(b.pending, headerID)
+			b.pendingMu.Unlock()
+		}()
+
+		// forward the request to gcs
+		b.forwardRequestToGcs(req)
+
+		// fetch the process ID from response
+		deadline := time.Now().Add(5 * time.Second)
```

Comment:
nit: Please declare the timeout at the beginning of the file so that we have a unified way for all the important constants.